### ISSUE TITLE
Upgrade to Asciidoctor.js 1.5.3

### ISSUE THREAD

**Issue by Mogztter** [asciidoctor/docgist#21]
Could you please test that everything is working fine with version 1.5.3.preview.2 ? If this is the case I will release a final version. 

Thanks :smile:

**Comment by mojavelinux** [comment]
To clarify, the version is 1.5.3-preview.2 (notice the hyphen). https://www.npmjs.com/package/asciidoctor.js

**Comment by nawroth** [comment]
I'm on it.

**Comment by nawroth** [comment]
There seems to be a difference in how attributes are handled. At least that's my guess because:

* Syntax highlighting doesn't kick in: http://nawroth.github.io/docgist/#_source_code_highlighting
* `imagesdir` isn't picked up for images: http://nawroth.github.io/docgist/?github-asciidoctor%2Fdocgist%2F%2Fgists%2Fimages.adoc

@Mogztter

**Comment by mojavelinux** [comment]
Yep, I raised this point in another issue. We need a hack in the build that
removes an obsolete hack for Opal. Attributes passed as a string aren't
getting split correctly.

**Comment by nawroth** [comment]
Plese update here when there's a fix.

**Comment by Mogztter** [comment]
Just released a `1.5.3-preview.3`.
This version should :zap: this :bug:

**Comment by mojavelinux** [comment]
Nice!

**Comment by nawroth** [comment]
The dist folder wasn't updated!

**Comment by mojavelinux** [comment]
That's definitely something we should automate.

**Comment by Mogztter** [comment]
Damn!
Yes manual releases are prone to error... we should definitely automate the
process! Automate all the things!!!

**Comment by Mogztter** [comment]
`1.5.3-preview.4` is out!

**Comment by nawroth** [comment]
I'm still seeing the same issue.
You can try it out at http://nawroth.github.io/docgist/

**Comment by Mogztter** [comment]
@nawroth With Opal 0.8.0 `doc.attributes.map` is empty (for an unknown reason) and you must use `doc.attributes.smap` instead. Or you can use `doc$attrs('name')` to stay out of Opal's "internal".

**Comment by nawroth** [comment]
`doc.attributes.smap` works. One difference I noticed is that running with `parse_header_only=true` will only return attributes that are actually in the document header, while it previously didn't mind a blank line between title and document attributes.

So I fixed the examples to work with it, for example:

* http://nawroth.github.io/docgist/?github-nawroth%2Fdocgist%2F%2Fgists%2Fimages.adoc
* http://nawroth.github.io/docgist/?github-nawroth%2Fdocgist%2F%2Fgists%2Fcodemirror.adoc

**Comment by Mogztter** [comment]
> One difference I noticed is that running with parse_header_only=true will only return attributes that are actually in the document header, while it previously didn't mind a blank line between title and document attributes.

I think this is related to https://github.com/asciidoctor/asciidoctor.js/issues/95#issuecomment-68576350

> doc.attributes.smap works.

Yes but you should use Hash's public methods instead: https://github.com/opal/opal/blob/master/opal/corelib/hash.rb

`smap` is an internal property and Opal can decide to modify it or remove it in future version :disappointed:

**Comment by nawroth** [comment]
@Mogztter I'll fix that, thanks for the links. The more important point is that 1.5.3 seems to work fine with DocGist now. :+1:

**Comment by Mogztter** [comment]
Yeah, thanks for your great work :smiley:

**Comment by nawroth** [comment]
Updated it to use `$has_key?` and `$fetch`.

Are there any unresolved issues with the previews or will 1.5.3 be out soon?

**Comment by Mogztter** [comment]
We're still discussing an "issue" on `attributes` [here](https://github.com/asciidoctor/asciidoctor.js/issues/96) and [here](https://github.com/asciidoctor/asciidoctor.js/issues/95) but I think the behavior is the same in Asciidoctor core (Ruby) and Asciidoctor.js... Hopefully @mojavelinux will clear up our questions and give us the green light to proceed.

So `1.5.3` should be out pretty soon!

**Comment by nawroth** [comment]
I couldn't find any more problems, so I pushed my changes + a lot around syntax highlighting to http://gist.asciidoctor.org/

Will do another update if there are any changes in the final `1.5.3`.

**Comment by mojavelinux** [comment]
>  it previously didn't mind a blank line between title and document attributes.

We should be strict about not permitting blank lines in the header, IMO.

**Comment by nawroth** [comment]
@mojavelinux Agreed.

**Comment by mojavelinux** [comment]
That's great news that you got it all working. Obviously, we still need to improve the usability of the APIs around document-level attributes and locking attributes, but that's for 1.6.0. What's important is that we have some really nice use cases to study now. As far as 1.5.3 is concerned, if we understand how to work with what is there, then we are ready for release.

**Comment by Mogztter** [comment]
Or better upgrade to https://github.com/asciidoctor/asciidoctor.js/releases/tag/v1.5.9
